stacks/serverless/dovetail-cdn-arranger.yml

```diff
@@ -0,0 +1,133 @@
+# stacks/serverless/dovetail-cdn-arranger.yml
+AWSTemplateFormatVersion: '2010-09-09'
+Description: Dovetail CDN Arranger lambda function
+Conditions:
+  IsProduction: !Equals [!Ref EnvironmentType, Production]
+Parameters:
+  CodeS3Bucket:
+    Type: String
+  CodeS3ObjectVersion:
+    Type: String
+  EnvironmentType:
+    Type: String
+  DovetailHost:
+    Type: AWS::SSM::Parameter::Value<String>
+  DovetailToken:
+    Type: AWS::SSM::Parameter::Value<String>
+  DynamodbTableArn:
```

Comment:
I still have some questions about how the nodejs code gets access to this.

- Should this be a table _name_ or _arn_?
- Is this table even in this account/region?
- Can we create a write-permission on this if it's not part of this account/region?

I seem to recall jumping through a bunch of hoops in the analytics lambda to get access to a data-production table.  Assuming a role linked to the other account and whatnot.  Wondering if there's an easier way here...

Reply:
I've only ever seen the JavaScript SDK expecting a DDB table **name**, so that is what I generally pass to Lambda functions in the `Environment`.

Where the table is shouldn't matter too much. IAM never cares about regions for access. For access across accounts, assuming a role with a trust policy is the correct way of doing it.

As for where the table lives, based on what I remember this table holds data that we would consider important and not want to lose, and probably even be a global table. That to me feels like it would live in the data-prod account

Reply:
Okay, so I'll rename the ENV to be more descriptive, change the ParameterStore param to be a table _name_ (not arn), and do some wildcard-ing in the IAM role (i think?).

And yes, this would ideally be a multi-region thing.  And probably the data account, though that's introducing some complexity with role-switching for all the lambdas using it.  I guess we'll get this one out there, then re-appraise how cross account DDB access works.